plugins/database/mssql/mssql_test.go

```diff
@@ -357,6 +357,53 @@ func TestDeleteUser(t *testing.T) {
 	assertCredsDoNotExist(t, connURL, dbUser, initPassword)
 }
 
+func TestDeleteUserContainedDB(t *testing.T) {
+	cleanup, connURL := mssqlhelper.PrepareMSSQLTestContainer(t)
+	defer cleanup()
+
+	dbUser := "vaultuser"
+	initPassword := "p4$sw0rd"
+
+	initReq := dbplugin.InitializeRequest{
+		Config: map[string]interface{}{
+			"connection_url": connURL,
+			"contained_db":   true,
+		},
+		VerifyConnection: true,
+	}
+
+	db := new()
+
+	dbtesting.AssertInitializeCircleCiTest(t, db, initReq)
+	defer dbtesting.AssertClose(t, db)
+
+	err := createTestMSSQLUser(connURL, dbUser, initPassword, testMSSQLLogin)
```

Comment:
I was experimenting a bit with this test, and it looks like `assertContainedDBCredsDoNotExist()` will always think that `vaultuser` doesn't exist, whether `db.DeleteUser` was called or not. So I think as part of "pretending" this is a contained db, we'll want to also create a user to go along with the login, so that there's something to be dropped. i.e.

```suggestion
	err := createTestMSSQLUser(connURL, dbUser, initPassword, testMSSQLContainedLogin)
```
where `testMSSQLContainedLogin` is something like:
```go
const testMSSQLContainedLogin = `
CREATE LOGIN [{{name}}] WITH PASSWORD = '{{password}}';
CREATE USER [{{name}}] FOR LOGIN [{{name}}];
`
```

Reply:
Updated in cf19eef